include/drivers/pwm.h

```diff
@@ -22,13 +23,45 @@
 #include <errno.h>
 #include <zephyr/types.h>
 #include <stddef.h>
+#include <sys/math_extras.h>
 #include <device.h>
 #include <dt-bindings/pwm/pwm.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @name PWM capture configuration flags
+ * @{
+ */
+
+/** PWM pin capture captures period. */
+#define PWM_CAPTURE_TYPE_PERIOD		(1 << 1)
+
+/** PWM pin capture captures pulse width. */
+#define PWM_CAPTURE_TYPE_PULSE		(1 << 2)
+
+/** PWM pin capture captures both period and pulse width. */
+#define PWM_CAPTURE_TYPE_BOTH		(PWM_CAPTURE_TYPE_PERIOD | \
+					 PWM_CAPTURE_TYPE_PULSE)
+
+/** @cond INTERNAL_HIDDEN */
+#define PWM_CAPTURE_TYPE_MASK		0x6
+/** @endcond */
+
+/* PWM pin capture captures a single period/pulse width. */
+#define PWM_CAPTURE_MODE_SINGLE		(0 << 3)
```

Comment:
There's some precedent for consistent use of `FOO_SHIFT` and `FOO_MASK` which would make this:

    #define PWM_CAPTURE_TYPE_SHIFT 1
    #define PWM_CAPTURE_TYPE_MASK (3U << PWM_CAPTURE_TYPE_SHIFT)
    #define PWM_CAPTURE_MODE_SHIFT 3
    #define PWM_CAPTURE_MODE_MASK (1U << PWM_CAPTURE_MODE_SHIFT)
    #define PWM_CAPTURE_MODE_SINGLE (0U << PWM_CAPTURE_MODE_SHIFT)
    #define PWM_CAPTURE_MODE_CONTINUOUS (1U << PWM_CAPTURE_MODE_SHIFT)

This reduces the magic numbers and makes it easier to change the bitfield assignments.

Reply:
Thanks, implemented.